couchbase/cluster.hxx

```diff
@@ -119,9 +119,9 @@ class cluster
     template<typename Handler>
     void close_bucket(const std::string& bucket_name, Handler&& handler)
     {
-        auto ptr = buckets_.find(bucket_name);
-        if (ptr != buckets_.end()) {
-            ptr->second->close();
+        auto bucket = find_bucket_by_name(bucket_name);
+        if (bucket) {
+            bucket->close();
```

Comment:
bucket->close only shuts down the bucket, but doesn't remove it from the buckets_ list.  This means that the actual object pointer would leak.

Reply:
right, thanks for noticing it